src/accounts/accounts.service.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  Injectable,
+  NotAcceptableException,
+  NotFoundException,
+} from '@nestjs/common';
+import { CreateTransactionDto } from 'src/transactions/dto/create-transaction.dto';
+import { CreateAccountDto } from './dto/create-account.dto';
+import { Account } from './entities/accounts.entity';
+import {
+  sendTransaction,
+  Transaction,
+} from 'src/transactions/entities/transactions.entity';
+import { SendTransactionDto } from 'src/transactions/dto/send-transaction.dto';
+import { sendMoneyValidation } from './functions/sendMoneyLogic';
+
+//data storage and retrieval (in memory)
+@Injectable()
+export class AccountsService {
+  private readonly accounts: Account[] = [];
+  private readonly transactions: (Transaction | sendTransaction)[] = [];
+
+  //getter to return transaction array to transactions service (for findAllTransactions)
+  get transactionsArray(): Transaction[] {
+    return this.transactions;
+  }
+
+  //creates a new account
+  create(createAccountDto: CreateAccountDto) {
+    const accountId: string = createAccountDto.id;
+    if (this.accounts.find((accounts) => accounts.id == accountId)) {
+      throw new NotAcceptableException(
+        'an account already exists for this id; cannot make duplicate accounts',
+      );
+    }
+    const newAccount = { ...createAccountDto };
+    this.accounts.push(newAccount);
```

Comment:
for this let's create a new array to avoid side effect of the accounts array 

Here is a reference!
https://github.com/ryanmcdermott/clean-code-javascript#avoid-side-effects-part-2

Reply:
Changed this in the latest commit